app/helpers/metadata_helper.rb

```diff
@@ -19,15 +19,39 @@ def metadata_url
 
   def format_title(title)
     return "Exercism" unless title
-
     "#{title} | Exercism"
   end
 
   def metadata
     @metadata ||=
       case namespace_name
+      when "admin"
+        { title: "Admin" }
       when "my"
-        nil
+        case controller_name
+        when "notifications"
+          { title: "Notifications" }
+        when "settings", "track_settings", "preferences"
+          { title: "Settings" }
+        when "reactions"
+          { title: "My Reactions" }
+        when "solutions"
+          { title: "My #{@track.title}/#{@exercise.title}" }
+        when "tracks"
+          case action_name
+          when "index"
+            { title: "My Tracks" }
+          when "show"
+            { title: @mentors ? "#{@track.title} Track Preview" : "My #{@track.title} Track" }
```

Comment:
I think this can be `"#{@track.title} Track" for both so you can remove the ternary. 

FYI, you could use `action_name == "show_not_joined"` to make your intent clearer.

Reply:
`action_name == "show_not_joined"` won't work because `show_not_joined` the show action isn't redirecting to `show_not_joined` but only returning that controller method.